src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2541,6 +2541,30 @@ public static void NJulianRuleTest(string posixRule, int dayNumber, int monthNum
             }
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public static void TimeZoneInfo_LocalZoneWithInvariantMode()
+        {
+            string hostTZId = TimeZoneInfo.Local.Id;
+
+            ProcessStartInfo psi = new ProcessStartInfo() {  UseShellExecute = false };
+            psi.Environment.Add("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", PlatformDetection.IsInvariantGlobalization ? "0" : "1");
+
+            RemoteExecutor.Invoke((tzId, hostIsRunningInInvariantMode) =>
+            {
+                bool hostInvariantMode = bool.Parse(hostIsRunningInInvariantMode);
```

Comment:
Why does the test check whether the process is already in invariant mode? Why not just launch the child process passing DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1 every time?

As it's written here, does it mean that you need to run on eg Alpine (where invariant mode is default) for the test to do its job of protecting your fix here?

Reply:
> Why does the test check whether the process is already in invariant mode? Why not just launch the child process passing DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1 every time?

We could do that, but we'll have to be careful in the future if decided to run the test with Invariant mode on.

> As it's written here, does it mean that you need to run on eg Alpine (where invariant mode is default) for the test to do its job of protecting your fix here?

For now, we don't need to do that, but nothing will prevent us to have this test run with invariant mode on. like copying/sharing this test to other tests which enable invariant mode.

Reply:
> We could do that, but we'll have to be careful in the future if decided to run the test with Invariant mode on.

I'm not following -- if we're forcing the child process to be in invariant mode, why would it matter whether someone runs all the tests in invariant mode? 

I am just wondering whether we should change this test so that when run on any distro/configuration, it tests the bug is fixed - not just Alpine.

Reply:
> I'm not following -- if we're forcing the child process to be in invariant mode, why would it matter whether someone runs all the tests in invariant mode?

The test is validating the time zone when running **with** invariant mode against when it is running **without** the invariant mode. If we the test running **without** invariant mode, we'll launch the child process **with** invariant mode and vice versa. So, it doesn't matter the configuration we are running on, the test should work and validate the concerned case.

> I am just wondering whether we should change this test so that when run on any distro/configuration, it tests the bug is fixed - not just Alpine.

That is exactly what the test is doing. This test should validate the concerned case regardless of the distros/configuration. It is not specific to Alpine.

Reply:
Ah, got it. thanks, I misread it backwards.